packages/wallet/src/KeyManagement/LedgerKeyAgent.ts

```diff
@@ -164,11 +170,68 @@ export class LedgerKeyAgent extends KeyAgentBase {
     });
   }
 
+  async signTransaction(initializedTxData: InitializeTxResult): Promise<Cardano.Signatures> {
+    try {
+      const deviceConnection = await LedgerKeyAgent.checkDeviceConnection(this.serializableData.communicationType, this.deviceConnection);
+      const dataForSigning = this.#prepareDataForSigning(initializedTxData);
+      const response = await deviceConnection.signTransaction(dataForSigning);
+      // TODO - derive witnes(es) path(s) and return Map<Ed25519PublicKey, Ed25519Signature>
+      return response.witnesses;
+    } catch (error: any) {
+      // eslint-disable-next-line unicorn/numeric-separators-style
+      if (error.code === 28169) {
+        throw new AuthenticationError('Transaction signing aborted', error);
+      }
+      throw new TransportError('Transport failed', error);
+    }
+    return response;
+  }
+
   async signBlob(): Promise<SignBlobResult> {
     throw new NotImplementedError('signBlob');
   }
 
   async exportRootPrivateKey(): Promise<Cardano.Bip32PrivateKey> {
     throw new NotImplementedError('Operation not supported!');
   }
+
+  async #prepareDataForSigning({ body, hash, inputSelection }: InitializeTxResult) {
+    // TODO - Isolate change outputs
+    // TODO - Prepare token bundle
+    const inputs = body.inputs.map((input) => ({
+      txHashHex: input.txId,
+      outputIndex: input.txIndex,
+      // TODO - Derivation Path - get from `ownSignatureKeyPaths`
+      path: utils.str_to_path("1852'/1815'/0'/0/0"),
+    }));
+
+    const outputs = body.outputs.map((output) => ({
+      amount: output.value.coins,
+      tokenBundle: output.value.assets || [], // TODO - Check & Calculate
+      destination: {
+        // TODO - remove address type declaration comments
+        // DEVICE_OWNED - for change address
+        // THIRD_PARTY - for regular output
+        type: TxOutputDestinationType.THIRD_PARTY,
```

Comment:
`if this.addresses.some(({address}) => address === output.address) then DEVICE_OWNED`

Reply:
Yes, I need to calculate outputs but also add additional parameters if address is change address ( + spendingPath
& stakingPath). I have TODO for that `// TODO - Isolate change outputs`

Reply:
Our staking path is always the same (2, 0) and spending path is available on objects in `this.addresses`

Reply:
aha, staking path is same `1852'/1815'/{accountIndex}'/0/0`. 
ouu yeah, spending path is in address obj